src/extension/app.tsx

```diff
@@ -43,14 +44,19 @@ chrome.runtime.sendMessage({ fetch: true }, function (response) {
             let tabId = entry.tabId;
             let id = json.envelope.impressionId;
             if (!(id in structured)) {
-                structured[id] = { envelope: json.envelope, events: [] };
-                structured[id]["envelope"].dateTime = entry.dateTime;
-                structured[id]["envelope"].summary = [];
+                if (json.envelope.sequenceNumber !== 0) {
+                    console.warn(`First payload for impression id ${id} has sequence number ${json.envelope.sequenceNumber}`);
+                    continue;
+                }
+                let metadata = json.metadata;
+                structured[id] = { metadata, events: [] };
+                structured[id].metadata.dateTime = entry.dateTime;
+                structured[id].metadata.summary = [];
```

Comment:
>metadata [](start = 31, length = 8)

Why change name from envelope to metadata?

Reply:
Because earlier this data was delivered from the client in the "envelope" property and, similarly, Sarvesh stored it in "envelope" here as well. 

Now, impression-level data is delivered in the "metadata" property on the first payload, so in visualization it's stored as "metadata" too. 

---
In reply to: [144541690](https://github.com/Microsoft/clarity-tools/pull/23#discussion_r144541690) [](ancestors = 144541690)